src/Compilers/CSharp/Portable/Symbols/Metadata/PE/SymbolFactory.cs

```diff
@@ -52,11 +52,16 @@ internal override Cci.PrimitiveTypeCode GetPrimitiveTypeCode(PEModuleSymbol modu
             return type.PrimitiveTypeCode;
         }
 
-        internal override bool IsVolatileModifierType(PEModuleSymbol moduleSymbol, TypeSymbol type)
+        internal override bool IsAcceptedVolatileModifierType(PEModuleSymbol moduleSymbol, TypeSymbol type)
         {
             return type.SpecialType == SpecialType.System_Runtime_CompilerServices_IsVolatile;
         }
 
+        internal override bool IsAcceptedIsConstModifierType(TypeSymbol type)
+        {
+            return type.IsWellKnownTypeIsConst();
+        }
```

Comment:
Can you please explain? caching what? this is not doing lookup, but rather checking the passed symbol for full name equality.

---
In reply to: [118763594](https://github.com/dotnet/roslyn/pull/19658#discussion_r118763594) [](ancestors = 118763594)